templates/prune.sh.j2

```diff
@@ -1,27 +1,39 @@
-#jinja2:lstrip_blocks: True
-#!/bin/bash
+#!{{ borgbackup_shell }}
+{{ ansible_managed | comment }}
 
-# This script is intended to run on a trusted management station to purge borg repo's in
-# append-only mode.
+{% macro repository(b, h) %}
+{% if b.type == 'hetzner'%}
+ssh://{{ b.user }}@{{ b.fqdn }}:23/./{{ b.home }}{{ b.pool }}/{{ h }}
+{%- else %}
+{{ b.user }}@{{ b.fqdn }}:{{ b.home }}{{ b.pool }}/{{ h }}
+{%- endif %}
+{% endmacro -%}
+
+# This script is intended to run on a trusted management station
+# to purge borg repos in append-only mode.
 # Don't put it on the backup server, it contains all borg secrets!
 
 {% for h in groups['all'] %}
-  {% if hostvars[h].borgbackup_required | default(True) -%}
```

Comment:
@dverhelst @stroobl 
I noticed my change on this line is failing the test, because I don't define a default.

In my opinion, this should default to false, as only hosts that require borg should be added to the prune script right?!

Reply:
It looks like this loop is not running correctly on molecule at least. I'm identifying the problem ...

Reply:
I'm confused myself :)

The following code temporary added to the prune.sh template:

```
# with default True
{% for h in groups['all'] %}
#  {{ h }} : {{ hostvars[h].borgbackup_required|default(True) }}
{% endfor %}
# with default False
{% for h in groups['all'] %}
#  {{ h }} : {{ hostvars[h].borgbackup_required|default(False) }}
{% endfor %}

```
Resulted in this:

```
# with default True
#  default-server-debian-10 : False
#  default-management-debian-10 : True
#  default-client-debian-10 : True
# with default False
#  default-server-debian-10 : False
#  default-management-debian-10 : False
#  default-client-debian-10 : False
```

@stroobl  care to chime in? This is 'odd' behaviour.

Reply:
I'm confused myself :)

The following code temporary added to the prune.sh template:

```
# with default True
{% for h in groups['all'] %}
#  {{ h }} : {{ hostvars[h].borgbackup_required|default(True) }}
{% endfor %}
# with default False
{% for h in groups['all'] %}
#  {{ h }} : {{ hostvars[h].borgbackup_required|default(False) }}
{% endfor %}

```
Resulted in this:

```
# with default True
#  default-server-debian-10 : False
#  default-management-debian-10 : True
#  default-client-debian-10 : True
# with default False
#  default-server-debian-10 : False
#  default-management-debian-10 : False
#  default-client-debian-10 : False
```

@stroobl  care to chime in? This is 'odd' behaviour.

Reply:
Not sure what @dverhelst is trying to prove here :), but I agree with @noplanman that the default should be false.

Reply:
@dverhelst That is expected behaviour, since the `borgbackup_required` variable is set to `False` in `default/molecule.yml` for the `borgbackup_servers` group.
So defaults apply to the client and management system, whereas the server is correctly `False` in both cases.

Or did I miss something you wanted to point out?

Reply:
@noplanman I re-tested and now I understand what's going wrong.
In the roles defaults/main.yml we have:
```
borgbackup_required: true
```
Now in prune.sh we put in the comment above here:
```
{%     if hostvars[h].borgbackup_required | default(False) and inventory_hostname == hostvars[h].borgbackup_management_station %}
```
This should be default(True) to match the role default. If you do that, you can remove your changes to molecule.yml as I requested in my review, since that didn't make sense to me. (but I didn't see this link at first)
Sorry for the confusion.

Reply:
@stroobl Ok, will revert and change this then 👍